scripts/tests/run_python_test.py

```diff
@@ -80,36 +80,52 @@ def DumpProgramOutputToQueue(thread_list: typing.List[threading.Thread], tag: st
                                            (f"[{tag}][\33[31mSTDERR\33[0m]").encode(), queue))
 
 
-@click.command()
-@click.option("--app", type=str, default=None, help='Local application to use, omit to use external apps, use a path for a specific binary or use a filename to search under the current matter checkout.')
-@click.option("--factoryreset", is_flag=True, help='Remove app config and repl configs (/tmp/chip* and /tmp/repl*) before running the tests.')
-@click.option("--app-params", type=str, default='', help='The extra parameters passed to the device.')
-@click.option("--script", type=click.Path(exists=True), default=FindBinaryPath("mobile-device-test.py"), help='Test script to use.')
-@click.argument("script-args", nargs=-1, type=str)
-def main(app: str, factoryreset: bool, app_params: str, script: str, script_args: typing.List[str]):
-    if factoryreset:
+def main():
+    parser = argparse.ArgumentParser(description='Run REPL Python Tests')
+
+    parser.add_argument('--app', type=str, default=None,
+                        help='Local application to use, omit to use external apps, use a path for a specific binary or use a filename to search under the current matter checkout.')
+    parser.add_argument('--factoryreset', action="store_true",
+                        help='Remove app config and repl configs (/tmp/chip* and /tmp/repl*) before running the tests.')
+    parser.add_argument('--app-args', type=str, default='',
+                        help='The extra parameters passed to the device side app.')
+    parser.add_argument('--script', type=str, default=None,
+                        help='Test script to use.')
+    parser.add_argument('--script-args', type=str, default='',
+                        help='Arguments for the REPL test script')
+
+    args = parser.parse_args()
+
+    if args.factoryreset:
         retcode = subprocess.call("rm -rf /tmp/chip* /tmp/repl*", shell=True)
         if retcode != 0:
             raise Exception("Failed to remove /tmp/chip* for factory reset.")
 
     log_queue = queue.Queue()
     log_cooking_threads = []
 
+    if (args.script == None):
+        args.script = os.path.dirname(os.path.realpath(
```

Comment:
This is not the same: it does not do a 'find' anymore but finds the direct path.